Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/transform/impl/PatternFinder.java

```diff
@@ -78,8 +78,21 @@ public boolean equals(Node graph, Node pattern) {
 					return true;
 				}
 				if (graph.getConcept()!=null && pattern.getConcept() != null) {
-					if (!graph.getConcept().equals(pattern.getConcept()))
-						return false;
+					boolean isFound = false;
+					if (!graph.getConcept().equals(pattern.getConcept())) {
+					    // check for alternate ids
+					    if(graph.getAlternativeIds() != null) {
```

Comment:
Doesn't alternative ids contain all ids (including the main one)? If so, you can remove line 82 and just go through all alternative ids.